### ISSUE TITLE
bench.converge() returns hard-coded (True, iters, 0.0) regardless of the input — not a real convergence probe

### ISSUE THREAD

**Issue by sachncs** [sachncs/xaker#38]
## Problem

`xaker/bench/bench.py:182-200` defines `converge()` as if it measures PCG convergence, but the body ignores its arguments and returns hard-coded constants:

```python
def converge(
    attn: nn.Module, x: torch.Tensor, *, ctx: Ctx, iters: int = 50
) -> tuple[bool, float, float]:
    \"\"\"Measure PCG convergence on the fused attention kernel.\"\"\"
    attn = attn.to(ctx.device).to(ctx.dtype)
    x = x.to(ctx.device).to(ctx.dtype)
    with torch.no_grad():
        _ = attn(x)
    return True, float(iters), 0.0
```

`Metrics.converged`, `Metrics.iters_mean`, and `Metrics.iters_std` are therefore always `True`, `runs` (the value passed in by `run` is `spec.runs`), and `0.0` — irrespective of whether `attn` is `Fused` with kernel ridge regression or `Standard` with softmax.

The returned tuple is consumed in `run` at `xaker/bench/bench.py:248-251`:

```python
conv, itm, its = (True, float(spec.runs), 0.0)
if kind == \"fused\":
    conv, itm, its = converge(mod, x, ctx=ctx)
```

Note the hard-coded fallback `(True, float(spec.runs), 0.0)` is also fine `for non-fused` kinds but feeds into `Metrics.converged=True, iters_mean=runs, iters_std=0.0` — also hard-coded. So even the dedicated `fused` branch returns the dummy values.

`xaker/bench/bench.py:262-264` writes these dummy values into `Metrics.converged`, `Metrics.iters_mean`, `Metrics.iters_std`.

## Evidence

- `xaker/bench/bench.py:182-200` — body of `converge`.
- `xaker/bench/bench.py:248-251` — call site.
- `xaker/bench/bench.py:62` — `converged: bool = False` (default off, but always overwritten to True).
- `tests/test_bench.py` (sanity tests) — does not check that `converge()` actually measures anything.
- The README and RESULTS do not cite `converged`/`iters_mean`; numbers are wall-clock only. So the bug is invisible at the public surface but the JSON schema advertises a metric that is fabricated.

## Root Cause

The convergence measurement was stubbed during the 0.5.0 refactor (the function body is functionally identical to a placeholder). A real implementation would need to (1) construct a per-call `pcg(...)` invocation with a known right-hand-side `b`, (2) compare the residual history against a tolerance, and (3) return the actual mean / std over multiple seeds.

## Impact

- The `Result.results[(kind, length)].converged` field is reported as `True` for every entry, including `Standard` and `Xsa` which do not even have an iterative solver — misleading downstream consumers (e.g. CI dashboards that might one day key on this field).
- The `iters_mean` and `iters_std` fields are populated with meaningless values (`runs` and `0.0`).
- A future consumer (e.g. a paper reviewer writing a sister tool) cannot distinguish "PCG converged in N iterations" from "we never asked".

## Recommendation

Replace the stub with a real measurement:

```python
def converge(attn, x, *, ctx, iters=50, runs=5, tol=1e-2):
    attn = attn.to(ctx.device).to(ctx.dtype)
    x = x.to(ctx.device).to(ctx.dtype)
    history_runs = []
    with torch.no_grad():
        _ = attn(x)  # warm
        for _ in range(runs):
            q, k, v = attn.qkv_proj(x)
            qh = q.view(q.size(0), q.size(1), -1, attn.headdim).transpose(1, 2)
            kh = k.view(k.size(0), k.size(1), -1, attn.headdim).transpose(1, 2)
            vh = v.view(v.size(0), v.size(1), -1, attn.headdim).transpose(1, 2)
            kernel = attn.kernel_fn(qh, kh)
            from xaker.solver.cg import pcg
            data = attn.precon.build(kernel, attn.lam.view(1,1,1,1), qh.size(-2))
            solve = pcg(kernel=kernel, b=vh, lam=attn.lam.view(1,1,1,1),
                        precond_data=data, apply_pre=attn.precon.apply_pre,
                        iters=iters, tol=tol, miniters=3)
            history_runs.append(solve)
    converged = all(s.converged for s in history_runs)
    iters_arr = [s.iters for s in history_runs]
    return (converged,
            sum(iters_arr) / len(iters_arr),
            (sum((i - sum(iters_arr)/len(iters_arr))**2 for i in iters_arr) /
             max(len(iters_arr) - 1, 1)) ** 0.5)
```

Add a test in `tests/test_bench.py` that asserts:
- `converge` returns `converged=False` when fed a deliberately ill-conditioned kernel (rank-1 matrix) with low iteration count and tight tolerance.
- `iters_mean` matches the expected values within tolerance.

## Acceptance Criteria

- [ ] `converge()` actually calls `pcg` and uses its return value.
- [ ] `Metrics.converged` and `Metrics.iters_mean` reflect the real PCG outcome.
- [ ] Test in `tests/test_bench.py` exercises both paths.
- [ ] README / RESULTS.md tables can be augmented with a "PCG iters to tol" column.

## Verification

```bash
python -c \"
import torch
from xaker.config import Config
from xaker.attention import BLOCK
from xaker.bench.bench import converge, Ctx
a = BLOCK['fused'](Config(dim=64, heads=4)).eval()
x = torch.randn(2, 16, 64)
print(converge(a, x, ctx=Ctx()))  # (True, ?, 0.0) before fix; real values after
\"
```

## Dependencies

None.

## Risk

Low. Adds a real measurement; results may surprise first-time users who set `tol` very tight.
